docs/external-module.format.md

```diff
@@ -0,0 +1,24 @@
+# External Module Format
+Typings are written in non-ambient external module format. Most of the time we refer it simply as external module format.
+
+This is the same format you will get if you run `tsc -d` without bundling the file (i.e., not using the `outFile` parameter).
+
+This document will gives you an overview of what is external module format and why the typings need to be written in this format.
```

Comment:
I'd be tempted to be a little more explicit here and lose mention of the `outFile` parameter; how about:

> The TypeScript Compiler uses the same format when generating a corresponding '.d.ts' file. So when you run `tsc` with [`tsc --declaration` / `tsc -d` compiler options](https://github.com/Microsoft/TypeScript/wiki/Compiler-Options) set the resulting '.d.ts' is in the external module format.)